Reset Factory HLD

Mohammedz93 opened the pull request:
Initial HLD of reset factory feature

ixodie committed:
Updating Contributors

Mohammedz93 committed:
Reset Factory HLD

Initial HLD of reset factory feature

ixodie committed:
Merge branch 'master' into js-Updating-Contributors

dgsudharsan commented on doc/reset_factory/ResetFactoryHLD.md:
```diff
@@ -0,0 +1,202 @@
+####  1.5.2. <a name='ConfigurationManagementRequirements'></a>Configuration and Management Requirements
+We will create a command line tool /usr/bin/reset-factory which should provide following functionality:
+
+1. default: reset configurations to factory default, clear all system logs and files.
```
Can you specify what files would exactly be cleared?

Mohammedz93 replied:
the list is already specified in the "Functional requirements"  and also in the "High-Level Design"

dgsudharsan commented on doc/reset_factory/ResetFactoryHLD.md:
```diff
@@ -0,0 +1,202 @@
+
+3. If not ("only-config" or “keep-basic”):
+
+* Delete bash , vim and python history files under "/home".
```
Will there be a warning to the user about these?

Mohammedz93 replied:
Sure

lucyhyde committed:
Merge branch 'master' into js-Updating-Contributors

Mohammedz93 committed:
HLD Fixes

Mohammedz93 committed:
Add error flow + fixes

Mohammedz93 committed:
Merge branch 'master' into reset_factory_hld

Mohammedz93 committed:
Typo fix

Mohammedz93 committed:
Merge branch 'reset_factory_hld' of https://github.com/Mohammedz93/SONiC into reset_factory_hld

Mohammedz93 committed:
Change diagram

Mohammedz93 committed:
Fix diagram

nkelapur committed:
Update EVPN_VXLAN_HLD.md (#1227)

Added Note in warm-reboot section as per issue

adyeung committed:
Update EVPN HLD on Fastboot limitation (#1233)

Update EVPN HLD on Fastboot limitation

dgsudharsan committed:
Collecting dump during SAI failure (#1212)

* Create dump_on_sai_failure.md
* This HLD is to describe the flow to collect useful dumps during SAI failures.

Mohammedz93 committed:
Merge branch 'master' into reset_factory_hld

Mohammedz93 committed:
HLD Fixes

ixodie committed:
Merge branch 'master' into js-Updating-Contributors

bingwang-ms committed:
Enhance show acl commands

bingwang-ms committed:
Add sample output for show acl rule

zhangyanzhao committed:
Merge pull request #1187 from ixodie/js-Updating-Contributors

Updating Contributors

kannankvs committed:
Updated Community Calendar for Menu (#1249)

Updated Community Calendar for Menu and better alignment

kannankvs committed:
Updated Supported Platform for Nvidia (#1250)

* Updated Supported Platform for Nvidia

Updated Supported Platform for Nvidia 2201

* Updated Ragile Platform

Updated Ragile Platform

kannankvs committed:
Updated supported platform for md file (#1251)

* Updated new platform for Barefoot Aurora 610

Updated new platform for Barefoot Aurora 610 and removed the Note

* Updated Nvidia and Tencent Platform

Updated Nvidia and Tencent Platform as per alphabetical order

* Updated few more new platforms

Updated few more new platforms for Netberg and Ragile platforms

zhangyanzhao committed:
Revert "Updating Contributors"

zhangyanzhao committed:
Merge pull request #1266 from sonic-net/revert-1187-js-Updating-Contributors

Revert "Updating Contributors"

kannankvs committed:
Fix for Contributing organization in Webpage (#1263)

* Fix for Contributing organization in Webpage

Fix for Contributing organization in Webpage and added a new contributor logo (clounix)

* Updated SONiC Members Logo

Updated SONiC Members Logo

kannankvs committed:
Updated Asic details for Ragile platform (#1270)

Updated Asic details for Ragile-b6910-64c platform as requested by Pettershao from Ragile.

dgsudharsan committed:
Config Reload Enhancement (#1203)

Design to address the issue sonic-net/sonic-buildimage#12071
Enhancing config reload to sequence the services and faster system initialization.

Immediately restart the critical services during config reload.
The non critical should be started only after all the ports are initialized.
Services can be configured to be started immediately or delayed. This can be using a field in FEATURE table.
The existing timers should be removed by this event driven approach.
This flow is applicable in case of all reboots (warm/fast/cold) as well as config reload.

stepanblyschak committed:
FIB Suppress Announcements of Routes Not Installed in HW (#1103)

This document describes a feedback mechanism that allows BGP not to advertise routes that haven't been programmed yet.

Signed-off-by: stepanblyschak <stepanb@nvidia.com>

kannankvs committed:
Updated partner logo

kannankvs committed:
Updated name change for Ragile platform

zhangyanzhao committed:
Merge pull request #1278 from kannankvs/logo_update_general

Updated partner logo

FuzailBrcm committed:
HLD update: Adding support for FPGA devices in PDDF (#1232)

HW vendors use FPGA devices in their platform design.

In one of the use cases, the FPGA is connected to CPU via I2C bus. Such FPGAs are similar to any other I2C device with I2C address and adapter where they are connected. We are calling it FPGAI2C.

The other use case is where the FPGA is connected to CPU via PCIe. The FPGA also acts as the I2C controller and other devices are controlled by it. We are calling this design as FPGAPCIe.

This update in the PDDF HLD deals with the support for these two kind of FPGAs.

Co-authored-by: Ying Xie <yxieca@users.noreply.github.com>

bingwang-ms committed:
Merge branch 'master' into improve_show_acl_commands

bingwang-ms committed:
Merge pull request #1261 from bingwang-ms/improve_show_acl_commands

[HLD] Enhance show acl commands to display status

zhangyanzhao committed:
Merge pull request #1279 from kannankvs/ragile_name_change

Updated name change for Ragile platform as requested by Ragile

liat-grozovik committed:
Merge branch 'master' into reset_factory_hld

kannankvs committed:
dos2unix run on shell script. And added # in first line of script (#1299)

bingwang-ms committed:
Improve HLD for show acl enhancement

bingwang-ms committed:
Merge pull request #1303 from bingwang-ms/improve_show_acl_cmd

[HLD] Improve HLD for show acl enhancement

oleksandrivantsiv committed:
Add DASH resources to CRM HLD. (#1265)

[DASH] Add DASH resources to CRM HLD

kannankvs committed:
Updated MoM  (#1300)

Updated MoM File

Updated MoM File

oleksandrivantsiv committed:
Static DNS HLD (#1262)

The access to the DNS in SONiC is organized via the resolver configuration file (/etc/resolv.conf). resolv.conf is the plain text file that contains the human-readable configuration. It is used across various subsystems in the SONiC to translate domain names into IP addresses. 

With the current implementation dynamic DNS configuration can be received from the DHCP server or static configuration can be set manually by the user. However, SONiC doesn't provide any protection for the static configuration. The configuration that is set by the user can be overwritten with the dynamic configuration at any time.

The proposed solution is to add support for static DNS configuration into Config DB. To be able to choose between dynamic and static DNS configurations `resolvconf` package.

rhalstea committed:
P4 Runtime App read path performance improvement HLD. (#1255)

* P4 Runtime App read path performance improvement HLD.

bhagatgit committed:
TSC Election Process (#1301)

* Initial Draft

* Remove production deployment example

* Remove the production deployment example to avoid confusion, the original text is self explanatory

* Add one SII entry for end customer profilieration.

* Add section for yearly weight

* Adjusted SII weight based on feedback

* Update SII for HLDs and PRs

* HLD and PR SII calculation to include both LFX SONiC and OCP SAI as discussed and agreed earlier

* Add min score, voting info

---------

Co-authored-by: Lihua Yuan <lihuay@live.com>
Co-authored-by: Adam Yeung <47261130+adyeung@users.noreply.github.com>
Co-authored-by: Lihua Yuan <lihuay@users.noreply.github.com>

mdanish-kh committed:
Delete config_reload duplicate image to fix `git clone` (#1302)

What I did
Delete duplicate image to fix git clone on NTFS-formatted hard-drives.

Why I did it
Git Clone of this repo will cause errors on a case-insensitive path filesystem (which is NTFS if you're on a Windows machine) because two files with same names but different casing exist on this repo.

'images/config_reload/Hostcfgd_flow_for_delayed_services.JPG'
'images/config_reload/Hostcfgd_flow_for_delayed_services.jpg'

The result is that, as Windows does not allow the two files to exist simultaneously, your git repo will never be in sync with the upstream SONiC repo as there will always be 1 local change (file deletion) that'll cause problems in your git workflow.

Even on macOS, you will have to reformat volume as case-insensitive/sensitive or make a new volume altogether.

lolyu committed:
[dualtor][active-active] Add doc for show command (#1304)

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>
Co-authored-by: Ying Xie <yxieca@users.noreply.github.com>

kannankvs committed:
Dell S6000 platform build image update (#1313)

Dell S6000 platform Build image update

updated supported platform page for html and md file for Dell S6000 platform Build image update.

kannankvs committed:
Updated the new member (#1314)

Updated the new member Beyond edge logo part of webpage

mihirpat1 committed:
Expected output for transceiver DOM CLI (#1306)

* Expected output for transceiver DOM CLI
---------

Signed-off-by: Mihir Patel <patelmi@microsoft.com>

lolyu committed:
[dualtor][active-active] Add `config mux mode` section (#1319)

Add config mux mode section to dualtor active-active HLD.

Signed-off-by: Longxiang Lyu lolv@microsoft.com

longhuan-cisco committed:
Move tx_disable/tx_disabled_channel to TRANSCEIVER_STATUS table (#1316)

Move tx_disable to TRANSCEIVER_STATUS table

oleksandrivantsiv committed:
Update static DNS HLD with minigraph changes (#1322)

Add flow to cover DNS configuration updates in docker containers.

Signed-off-by: Oleksandr Ivantsiv <oivantsiv@nvidia.com>
Co-authored-by: Liat Grozovik <44433539+liat-grozovik@users.noreply.github.com>

liat-grozovik committed:
Merge branch 'master' into reset_factory_hld
